stacktrace.go

```diff
@@ -350,14 +353,51 @@ func callerFunctionName() string {
 	return baseName(callerFrame.Function)
 }
 
+func goMajorMinorVersion() (int, int, error) {
```

Comment:
fwiw I wrote this for something internally.

```go
// version gets the current version of Go from the runtime. The version may be
// in the forms:
//  - go1.17
//  - vdevel go1.17-9a8788c99d Tue Nov 16 10:53:28 2021 +0100
//
// Due to the stark difference in values, the version needs to be located and
// extracted.
func version() (string, error) {
	v := runtime.Version()
	for _, f := range strings.Fields(v) {
		const prefix = "go"
		if !strings.HasPrefix(f, prefix) {
			continue
		}
		f = f[len(prefix):]
		if i := strings.IndexRune(f, '-'); i != -1 {
			f = f[:i]
		}
		return f, nil
	}
	return "", fmt.Errorf("valid go version not found in %q", v)
}
```

It might be nice to avoid regular expressions where possible

Reply:
I actually rewrote it as

```go
func version() (string, error) {
	releaseTags := build.Default.ReleaseTags
	if len(releaseTags) == 0 {
		return "", errors.New("no release tags")
	}
	// The last release tag is presumed to be the current release. The current
	// release will be in the form 'go1.2.3'.
	//
	// See https://pkg.go.dev/go/build#Context
	return strings.TrimPrefix(releaseTags[len(releaseTags)-1], "go"), nil
}
```

Reply:
Then I guess you could extract the version like

https://go.dev/play/p/qTzf10VVF8k

```go
package main

import "fmt"

func main() {
	var major, minor int
	if _, err := fmt.Sscanf("go1.2.3", "go%d.%d", &major, &minor); err != nil {
		panic(err)
	}
	fmt.Println(major, minor)
}
```

So, altogether:

https://go.dev/play/p/U_DwYsnV9Nv

```go
package main

import (
	"errors"
	"fmt"
	"go/build"
)

func version() (major, minor int, err error) {
	releaseTags := build.Default.ReleaseTags
	if len(releaseTags) == 0 {
		return -1, -1, errors.New("no release tags")
	}
	// The last release tag is presumed to be the current release. The current
	// release will be in the form 'go1.2.3'.
	//
	// See https://pkg.go.dev/go/build#Context
	if _, err := fmt.Sscanf(releaseTags[len(releaseTags)-1], "go%d.%d", &major, &minor); err != nil {
		return -1, -1, fmt.Errorf("sscanf: %w", err)
	}
	return major, minor, nil
}

func main() {
	fmt.Println(version())
}
```

Reply:
Thanks!